File: python_projects/mario/mario.py
from cs50 import get_int


def main():
    '''ask for height,check user input'''
    pyramid_height = get_height()
    print_pyramid(pyramid_height)


def get_height():
    while True:
        height = get_int("Height: ")
        if height <= 8 and height > 0:
            break
    return height
    

def print_pyramid(pyramid_height):
    count_spaces = pyramid_height - 1
    count_blocks = 1
    for i in range(pyramid_height):
        for j in range(count_spaces):
            print(" ", end="")
        count_spaces -= 1
        for k in range(count_blocks):
            print("#", end="")
        print()
        count_blocks += 1


if __name__ == "__main__":
    main()

File: c_projects/mario.c
#include <cs50.h>
#include <stdio.h>

int heightFromUserInput(void);
void printPyramid(int, int, int);
int main(void)
{
    
    int height = heightFromUserInput();
    int countSpaces = height - 1;
    int countHeightOfPyramid = 1;
    
    printPyramid(countSpaces, height, countHeightOfPyramid);
    
}

int heightFromUserInput(void)
{
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (height > 8 || height < 1);
    
    return height;

}

void printPyramid(int countSpaces, int height, int countHeightOfPyramid)
{
    for (int i = 0; i < height; i++)
    {
        
        for (int d = 0; d < countSpaces; d++)
        {
            printf(" "); 
        
        }
        
        countSpaces--;
        
        for (int j = 0; j < countHeightOfPyramid; j++)
        {
            printf("#");
        }
      
        countHeightOfPyramid++;
    
        printf("\n");
    }
}

